File: logstash-core/lib/logstash/instrument/metric_type/counter.rb
# encoding: utf-8
require "logstash/instrument/metric_type/base"
require "concurrent"

module LogStash module Instrument module MetricType
  class Counter < org.logstash.instrument.metrics.Counter
    def execute(action, value = 1)
      self.send(action, value)
    end

    # We don't want this ruby style method in java-land
    def value
      self.getValue()
    end
  end
end; end; end

File: logstash-core/src/main/java/org/logstash/instrument/metrics/Counter.java
package org.logstash.instrument.metrics;

import java.util.concurrent.atomic.LongAdder;

/**
 * Created by andrewvc on 5/30/17.
 */
public class Counter {
    private final LongAdder value = new LongAdder();

    public Counter(long initialValue) {
        this.value.add(initialValue);
    }

    public void increment(long v) {
        this.value.add(v);
    }

    public void increment() {
        this.increment(1);
    }

    public void decrement(long v) {
        this.value.add(-v);
    }

    public void decrement() {
        this.decrement(1);
    }

    public long get() {
        return this.value.longValue();
    }
}
